### PR TITLE
feat(e2ei): respect E2EI during login and MLS client creation (WPB-5851)

### PR DESCRIPTION
**Pull request by AndroidBob**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5851" title="WPB-5851" target="_blank">TaskWPB-5851</a>  [Android] Respect E2EI when Creating New MLS Client
  </td></table>
  <br />
 

Cherry pick from the original PR: 
- #2621

---- 

 ⚠️ Conflicts during cherry-pick:
app/src/main/kotlin/com/wire/android/feature/e2ei/OAuthUseCase.kt

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like 
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Respecting E2EI state during MLS client registration and login.

#### Test Coverage (Optional)
WIP
- [ ] I have added automated test to this contribution

#### How to Test

Have a team that has e2ei enabled, during login the user required to enroll certificate first.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. .

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2776](https://10.10.124.134/job/AR-build-pipeline/2776/) **failed**.

**Review by borichellow — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -229,6 +229,15 @@
...
 
+        <!--
+        We make a custom Firebase init, because the app selects the Firebase project at a runtime,
+        so we need to remove default FirebaseInitProvider and create our custom FirebaseInitializer.
+        -->
+        <provider
+                android:name="com.google.firebase.provider.FirebaseInitProvider"
+                android:authorities="${applicationId}.firebaseinitprovider"
+                tools:node="remove" />
+
```
Comment (ohassine) [inline app/src/main/AndroidManifest.xml]: This duplicates  line 242

**Comment by AndroidBob** [comment]
Build [2911](https://10.10.124.134/job/AR-build-pipeline/2911/) **failed**.

**Review by ohassine — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [2923](https://10.10.124.134/job/AR-build-pipeline/2923/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(8826600)(8826600)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(8826600)(8826600)-staging-compat.apk)

**Pull request merged**